sources/packages/web/src/views/aest/institution/Programs.vue

```diff
@@ -98,26 +139,58 @@ export default defineComponent({
       {} as PaginatedResults<EducationProgramsSummary>,
     );
     const searchProgramName = ref("");
+    const searchLocationName = ref("");
     const currentPage = ref();
     const currentPageLimit = ref();
     const loading = ref(true);
-
+    const programStatusItems = ref([
+      ProgramStatus.Approved,
+      ProgramStatus.Pending,
+      ProgramStatus.Declined,
+      INACTIVE_PROGRAM,
+    ]);
+    const searchProgramStatus = ref([
+      ProgramStatus.Approved,
+      ProgramStatus.Pending,
+      ProgramStatus.Declined,
+      INACTIVE_PROGRAM,
+    ]);
     const getProgramsSummaryList = async (
       institutionId: number,
       rowsPerPage: number,
       page: number,
-      programName: string,
       sortColumn?: ProgramSummaryFields,
       sortOrder?: DataTableSortByOrder,
     ) => {
       try {
         loading.value = true;
-        searchProgramName.value = programName;
+        const statusSearchList = JSON.parse(
+          JSON.stringify(searchProgramStatus.value),
+        );
+        const searchInactiveProgram = statusSearchList.indexOf("Inactive") > -1;
+        if (searchInactiveProgram) {
+          statusSearchList.splice(statusSearchList.indexOf("Inactive"), 1);
+        }
+        let searchCriteria: Record<string, string | string[] | boolean>;
+        if (statusSearchList.length) {
+          searchCriteria = {
+            programNameSearch: searchProgramName.value,
+            locationNameSearch: searchLocationName.value,
+            statusSearch: statusSearchList,
+            inactiveProgramSearch: searchInactiveProgram,
+          };
+        } else {
+          searchCriteria = {
+            programNameSearch: searchProgramName.value,
+            locationNameSearch: searchLocationName.value,
+            inactiveProgramSearch: searchInactiveProgram,
+          };
+        }
```

Comment:
Can it be refactored as below? The idea is to execute more array checks than string manipulations.
```ts
const statusSearchList = searchProgramStatus.value.filter(
    (searchItem) => searchItem !== INACTIVE_PROGRAM,
  );
  const searchInactiveProgram = searchProgramStatus.value.some(
    (searchItem) => searchItem === INACTIVE_PROGRAM,
  );
  let searchCriteria: Record<string, string | string[] | boolean> = {
    programNameSearch: searchProgramName.value,
    locationNameSearch: searchLocationName.value,
    statusSearch: statusSearchList,
    inactiveProgramSearch: searchInactiveProgram,
  };
  if (statusSearchList.length) {
    searchCriteria.statusSearch = statusSearchList.toString(); // Check if we need JSON.stringify.
  }
```

Reply:
No conversion needed here as `filter` returns a string array.